server/controllers/entriesController.js

```diff
@@ -65,13 +65,92 @@ export const viewEntries = async (req, res) => {
   }
 };
 export const viewSpecificEntry = async (req, res) => {
-
+  try {
+    const { userId } = req.body;
+    const { entryId } = req.params;
+    const { rows } = await database.viewSpecificEntry(userId, entryId);
+    let retrievedEntry = rows;
+    retrievedEntry = retrievedEntry.map((entryInfo) => ({
+      createdOn: entryInfo.createddate,
+      entryId: entryInfo.entryid,
+      title: entryInfo.title,
+      description: entryInfo.description,
+    }));
+    if (retrievedEntry.length) {
+      res.status(200).json({
+        status: res.statusCode,
+        data: {
+          userId,
+          entry: retrievedEntry,
+        },
+      });
+    }
+    res.status(404).json({
+      status: res.statusCode,
+      error: 'Not Found!',
+    });
+  } catch (err) {
+    if (err.routine === 'string_to_uuid') {
+      res.status(404).json({
+        status: res.statusCode,
+        error: 'Not Found!',
+      });
+    }
+    res.status(500).json({
+      status: res.statusCode,
+      error: err,
+    });
+  }
```

Comment:
@T2Wil great work implementing this features, But this block of code is repeating itself in many areas, can you create one which is reusable for code maintainability.

Reply:
@mcaleb808 
Solved!
I have created a new file containing instructions to handle errors.